### PR TITLE
feat: pass stderr

### LINKED ISSUE #63: [feature] Pass stderr from executed command when running gum spin
Hey everyone!

I am running [borg backup](https://www.borgbackup.org/) via `gum spin` in a bash script and want to write resulting output from borg to a log file. I noticed that gum seems to pass _stdout_ but swallows _stderr_.
Borg passes all output through _stderr_ by default as documented [here](https://borgbackup.readthedocs.io/en/stable/usage/general.html#logging), which makes it difficult to collect logs when running inside of `gum spin`.

In my opinion it would make sense to introduce a new CLI option called `--show-error` that can be used like `--show-output` and passes _stderr_ instead.

### PR DESCRIPTION
**Pull request by aa-ko**
Pass stderr from the command run by gum spin analog to how stdout is passed.

This fixes #63.

### CODE REVIEW

File: spin/command.go [hunk 1]
```diff
@@ -31,7 +31,11 @@ func (o Options) Run() error {
 	}
 
 	if o.ShowOutput {
-		fmt.Print(m.output)
+		fmt.Fprint(os.Stdout, m.out_std)
```
Comment (maaslalani) [inline spin/command.go:34]:
```suggestion
		fmt.Fprint(os.Stdout, m.stdout)
```

File: spin/command.go [hunk 2]
```diff
@@ -31,7 +31,11 @@ func (o Options) Run() error {
...
+	}
+
+	if o.ShowError {
+		fmt.Fprint(os.Stderr, m.out_err)
```
Comment (maaslalani) [inline spin/command.go:38]:
```suggestion
		fmt.Fprint(os.Stderr, m.stderr)
```

File: spin/spin.go [hunk 3]
```diff
@@ -28,13 +29,15 @@ type model struct {
 	command []string
 	aborted bool
 
-	status int
-	output string
+	status  int
+	out_std string
```
Comment (maaslalani) [inline spin/spin.go:33]:
```suggestion
	stdout string
```

File: spin/spin.go [hunk 4]
```diff
@@ -28,13 +29,15 @@ type model struct {
...
+	out_err string
```
Comment (maaslalani) [inline spin/spin.go:34]:
```suggestion
	stderr string
```

File: spin/spin.go [hunk 5]
```diff
@@ -28,13 +29,15 @@ type model struct {
...
 }
 
 type finishCommandMsg struct {
-	output string
-	status int
+	out_std string
```
Comment (maaslalani) [inline spin/spin.go:38]:
```suggestion
	stdout string
```

File: spin/spin.go [hunk 6]
```diff
@@ -28,13 +29,15 @@ type model struct {
...
+	out_err string
```
Comment (maaslalani) [inline spin/spin.go:39]:
```suggestion
	stderr string
```

File: spin/spin.go [hunk 7]
```diff
@@ -53,8 +61,9 @@ func commandStart(command []string) tea.Cmd {
 		}
 
 		return finishCommandMsg{
-			output: string(out),
-			status: status,
+			out_std: outbuf.String(),
```
Comment (maaslalani) [inline spin/spin.go:64]:
```suggestion
			stdout: outbuf.String(),
```

File: spin/spin.go [hunk 8]
```diff
@@ -71,7 +80,8 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	var cmd tea.Cmd
 	switch msg := msg.(type) {
 	case finishCommandMsg:
-		m.output = msg.output
+		m.out_std = msg.out_std
```
Comment (maaslalani) [inline spin/spin.go:83]:
```suggestion
		m.stdout = msg.stdout
```

File: spin/spin.go [hunk 9]
```diff
@@ -71,7 +80,8 @@ func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
...
+		m.out_err = msg.out_err
```
Comment (maaslalani) [inline spin/spin.go:84]:
```suggestion
		m.stderr = msg.stderr
```

File: spin/spin.go [hunk 10]
```diff
@@ -44,7 +47,12 @@ func commandStart(command []string) tea.Cmd {
 			args = command[1:]
 		}
 		cmd := exec.Command(command[0], args...) //nolint:gosec
-		out, _ := cmd.Output()
+
+		var outbuf, errbuf strings.Builder
+		cmd.Stdout = &outbuf
+		cmd.Stderr = &errbuf
+
+		cmd.Run()
```
Comment (maaslalani) [inline spin/spin.go:55]:
```suggestion
		_ = cmd.Run()
```

**Comment by maaslalani** [comment]
Hey, @aa-ko! Great work on this. I removed the `--show-error` flag and simply output the `stderr` when `--show-output` is passed as people can redirect the `stderr` where they want.

```
gum spin --show-output -- <action> 2>/dev/null
```

If they don't want to see the `stderr` of the command.

**Comment by aa-ko** [comment]
Hey @maaslalani, thanks for the quick response and the little fix!

**Pull request merged**